src/main/java/Main.java

```diff
@@ -2,19 +2,49 @@
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.ArrayList;
 import java.util.List;
 import java.util.stream.Collectors;
 
 public class Main {
+
+    private static final Integer N_INDEXING_FILES = 100;
+
+    private static <T> List<List<T>> partition(List<T> collection, int partitionSize){
```

Comment:
Using this method as private static makes it difficult to be tested. Moreover the responsability of partitioning a list does not seem to be part of the main class.